### ISSUE TITLE
Update LayerStack documentation

### ISSUE THREAD

**Issue by nvidia-jomiller** [PixarAnimationStudios/USD#1855]
### Description of Issue
As previously discussed in https://github.com/PixarAnimationStudios/USD/issues/1778, the [Usd Documentation](https://graphics.pixar.com/usd/release/glossary.html#) could use an update to further describe `Reference`s with a `LayerOffset` in the `LayerStack`.

Some of the important details to update the documentation with would be the following:
* The role that a `LayerOffset` plays in the identifier of a `Reference` in the `LayerStack`
* How `TimeCodesPerSecond` are automatically computed and stored as `LayerOffset`s in the `LayerStack`
* Multiple sub layers each with different `TimeCodesPerSecond` that add / delete the same `Reference` (implicit `LayerOffset`)
* Multiple sub layers that add / delete the same `Reference` with explicitly authored `LayerOffset`s

The most accessible place for this information to live would probably be in the [LayerStack](https://graphics.pixar.com/usd/release/glossary.html#usdglossary-layerstack) section of the documentation. The `LayerStack` example could be extended to include `LayerOffset`s and the behavior to be expected when deleting one of those `Reference`. Since this information relates to multiple concepts it might be convenient to make sure the `Reference` and `LayerOffset` sections link to the `LayerStack` example.

**Comment by spiffmon** [comment]
Thanks for filing, @nvidia-jomiller . Just wanted to clarify on this one:

> Adding a Reference to multiple Layers with different LayerOffsets (i.e different TimeCodesPerSecond)

... are you saying we're adding a reference to the **same** layer in multiple layers in a stack, with different explicitly authored LayerOffsets, or just referencing different layers that each have different TCPS, _without_ explicit LayerOffsets?  In the first case, the referenced layer's TCPS is irrelevant (I think, though I should follow up with Corey to confirm), and in the second, LayerOffsets are irrelevant to reference identity because they are all auto-computed.

**Comment by nvidia-jomiller** [comment]
You're right @spiffmon, the wording there was a little unclear. I've updated the description to hopefully clarify.

My intent was to get documentation to also highlight the differences between implicit and explicit `LayerOffset`s for a `Reference`. But to specifically answer you're question, it was more to do with the second. While implicit (autocomputed) `LayerOffset`s are no longer playing a role in the identity of the `Reference` I still think it would be good to mention that in comparison to an explicit `LayerOffset`.

**Comment by jilliene** [comment]
Filed as internal issue #USD-7363
